pkgs/development/compilers/llvm/common/lld/default.nix

```diff
@@ -19,21 +19,23 @@ let
   pname = "lld";
   src' =
     if monorepoSrc != null then
-      runCommand "lld-src-${version}" {} ''
+      runCommand "lld-src-${version}" {} (''
         mkdir -p "$out"
+      '' + lib.optionalString (lib.versionAtLeast release_version "14") ''
         cp -r ${monorepoSrc}/cmake "$out"
+      '' + ''
         cp -r ${monorepoSrc}/${pname} "$out"
         mkdir -p "$out/libunwind"
         cp -r ${monorepoSrc}/libunwind/include "$out/libunwind"
         mkdir -p "$out/llvm"
-      '' else src;
+      '') else src;
 
   postPatch = lib.optionalString (lib.versions.major release_version == "12") ''
     substituteInPlace MachO/CMakeLists.txt --replace \
       '(''${LLVM_MAIN_SRC_DIR}/' '('
     mkdir -p libunwind/include
     tar -xf "${libunwind.src}" --wildcards -C libunwind/include --strip-components=2 "libunwind-*/include/"
-  '' + lib.optionalString (lib.versions.major release_version == "13" && stdenv.hostPlatform.isDarwin) ''
+  '' + lib.optionalString (lib.versions.major release_version == "13") ''
     substituteInPlace MachO/CMakeLists.txt --replace \
       '(''${LLVM_MAIN_SRC_DIR}/' '(../'
   '';
```

Comment:
I was curious why this was needed; just for posterity:
  - lld 13 (and older) use `llvm-config` to find `LLVM_MAIN_SRC_DIR` ([here](https://github.com/llvm/llvm-project/blame/d7b669b3a30345cfcdb2fde2af6f48aa4b94845d/lld/CMakeLists.txt#L9-L31)); in our case this doesn't work because `llvm-config --src-root` doesn't actually exist (it's the tmp dir used to build the llvm derivation). Hence the `substituteInPlace` for lld 12 and lld 13
  - lld 14 had John's patches for standalone builds ([1](https://github.com/llvm/llvm-project/commit/a1da5f3c2d65b198b79b90f9dbd2037837623328), [2](https://github.com/llvm/llvm-project/commit/319f4b2d52e31b000db75a0a2484b5f2ab90534a)) that use `../llvm` for `LLVM_MAIN_SRC_DIR` when `llvm-config` is not specified; we still specify `LLVM_CONFIG_PATH` for lld ≤14 though so [`fix-root-src-dir.patch`](https://github.com/NixOS/nixpkgs/blob/7df8f7c90d5d3b3a234657eb3daa955b550531b9/pkgs/development/compilers/llvm/14/lld/fix-root-src-dir.patch) is our workaround

Reply:
> For some reason, the component is built from a different directory in LLVM 12 and 13 so we can't unify postPatch between 12 and 13, unfortunately.

re the discrepancy between 12 (`LLVM_MAIN_SRC_DIR = .`) and 13 (`LLVM_MAIN_SRC_DIR = ../`): this is just due to lld 13 using `monorepoSrc` and lld 12 not

for lld 13 the directory structure looks like this:
```console
source
├── libunwind
│   ├── CMakeLists.txt
│   ├── LICENSE.TXT
│   ├── cmake
│   ├── docs
│   ├── include
│   ├── src
│   └── test
└── lld  <--- is PWD
    └── ... <snipped>
```

and for lld 12:
```console
lld-12.0.1.src  <--- is PWD (just the lld dir)
├── libunwind
│   └── include
└── ... <snipped; lld dir contents>
```

---

@sternenseemann: if/when we switch to using `monorepoSrc` for `llvmPackages_12` we can paper over the difference and unify the `postPatch` commands (i.e. just let the `runCommand` that produces `src` place `libunwind` at `lld/..` in both cases)

Reply:
Thanks, I've updated the commit message!